android/android_components.md

```diff
@@ -0,0 +1,24 @@
+# Building android-components #
+
+## Preparation ##
+
+1. Install Android Studio if it is not yet installed. This will also install a JDK.
+2. Find out where Android Studio installed the JDK by:
+    1. Opening Android Studio.
+    2. Selecting "Configure."
+    3. Selecting "Default Project Structure..."
+    4. You should now see the Android JDK location. (On OS X, mine is `/Applications/Android Studio.app/Contents/jre/jdk/Contents/Home`.)
+3. Set the environment variable `JAVA_HOME` to the path of the JDK. I did this on OS X by adding the line `# export JAVA_HOME="/Applications/Android Studio.app/Contents/jre/jdk/Contents/Home"` to `.bash_profile`.
```

Comment:
Interesting! I've actually have told my team members to install Java independently but it's a neat approach to let Android Studio manage it instead. One downside I see is that if the Android Studio path changes, the configuration may break and devs might not know why.

One question: if you `cd` into the `a-c` directory and run a command line command using Java, like `./gradlew tasks`, does the build run successfully? I wonder if we need to add java to the path as well. We often use command line tooling for git push hooks so ideally any config we suggest will work for those too!

I'd just test this locally but I don't want to break my existing build 😁 so that'll need to wait!

Reply:
> One question: if you cd into the a-c directory and run a command line command using Java, like ./gradlew tasks, does the build run successfully?

When I try building at the command line (without setting `JAVA_HOME`) on OS X, I get a message that I need to install a JDK.

> One question: if you cd into the a-c directory and run a command line command using Java, like ./gradlew tasks, does the build run successfully?

I think anyone who works at the command line would know they need to install java and have it on the path.